### ISSUE TITLE
Connecting to pgpool with SSL segfaults

### ISSUE THREAD

**Issue by zathey** [pgbouncer/pgbouncer#133]
Running psql through PgBouncer to Postgres using SSL works. Running psql through pgpool to Postgres using SSL works. Running psql through PgBouncer to pgpool with `server_tls_sslmode` set to disable works but it does not work if `server_tls_sslmode` is set to require or verify-ca. I do not have this problem on PgBouncer 1.7, only 1.7.1 and 1.7.2 have this issue.

psql to pgpool works with SSL

```
client$ ./psql -h 192.168.99.100 -p 9999 -U postgres
Password for user postgres: 
psql (9.4.5, server 9.2.15)
SSL connection (protocol: TLSv1, cipher: AES256-SHA, bits: 256, compression: off)
Type "help" for help.

postgres=# 
```

psql to PgBouncer does not work

```
client$ ./psql -h 192.168.99.100 -p 6432 -U postgres
Password for user postgres: 
psql: server closed the connection unexpectedly
    This probably means the server terminated abnormally
    before or while processing the request.
```

PgBouncer log

```
server# su pgbouncer -c 'pgbouncer /etc/pgbouncer/pgbouncer.ini'
2016-04-27 21:10:59.711 2549 DEBUG parse_ini_file: 'verbose' = '2' ok:1
2016-04-27 21:10:59.711 2549 DEBUG loading auth_file: "/etc/pgbouncer/userlist.txt"
2016-04-27 21:10:59.712 2549 NOISE event: 136, SBuf: 200, PgSocket: 392, IOBuf: 4108
2016-04-27 21:10:59.712 2549 LOG File descriptor limit: 1024 (H:4096), max_client_conn: 100, max fds possible: 130
2016-04-27 21:10:59.712 2549 DEBUG pktbuf_dynamic(128): 0x7b8ee0
2016-04-27 21:10:59.712 2549 DEBUG make_room(0x7b8ee0, 9): realloc newlen=256
2016-04-27 21:10:59.712 2549 DEBUG pktbuf_dynamic(128): 0x7be130
2016-04-27 21:10:59.712 2549 NOISE connect(4, unix:/tmp/.s.PGSQL.6432) = No such file or directory
2016-04-27 21:10:59.712 2549 DEBUG adns_create_context: evdns2
2016-04-27 21:10:59.712 2549 DEBUG add_listen: 192.168.99.100:6432
2016-04-27 21:10:59.712 2549 NOISE old TCP_DEFER_ACCEPT on 8 = 0
2016-04-27 21:10:59.712 2549 NOISE install TCP_DEFER_ACCEPT on 8
2016-04-27 21:10:59.712 2549 LOG listening on 192.168.99.100:6432
2016-04-27 21:10:59.712 2549 DEBUG add_listen: unix:/tmp/.s.PGSQL.6432
2016-04-27 21:10:59.712 2549 LOG listening on unix:/tmp/.s.PGSQL.6432
2016-04-27 21:10:59.712 2549 LOG process up: pgbouncer 1.7.2, libevent 2.0.21-stable (epoll), adns: evdns2, tls: OpenSSL 1.0.1e-fips 11 Feb 2013
2016-04-27 21:11:02.730 2549 NOISE new fd from accept=10
2016-04-27 21:11:02.730 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:02.730 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 pkt='!' len=8
2016-04-27 21:11:02.730 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 C: req SSL
2016-04-27 21:11:02.730 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 P: nak
2016-04-27 21:11:02.731 2549 NOISE resync: done=8, parse=8, recv=8
2016-04-27 21:11:02.731 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 P: got connection: 192.168.99.1:52239 -> 192.168.99.1:6432
2016-04-27 21:11:02.731 2549 NOISE safe_accept(8) = Resource temporarily unavailable
2016-04-27 21:11:02.731 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:02.731 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 pkt='!' len=84
2016-04-27 21:11:02.731 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 got var: user=postgres
2016-04-27 21:11:02.731 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 got var: database=postgres
2016-04-27 21:11:02.731 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 using application_name: psql
2016-04-27 21:11:02.731 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52239 got var: client_encoding=UTF8
2016-04-27 21:11:02.731 2549 LOG C-0x7cbf60: postgres/postgres@192.168.99.1:52239 login attempt: db=postgres user=postgres tls=no
2016-04-27 21:11:02.731 2549 NOISE resync: done=84, parse=84, recv=84
2016-04-27 21:11:02.731 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:02.731 2549 LOG C-0x7cbf60: postgres/postgres@192.168.99.1:52239 closing because: client unexpected eof (age=0)
2016-04-27 21:11:04.433 2549 NOISE new fd from accept=10
2016-04-27 21:11:04.433 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:04.433 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 pkt='!' len=8
2016-04-27 21:11:04.433 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 C: req SSL
2016-04-27 21:11:04.433 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 P: nak
2016-04-27 21:11:04.438 2549 NOISE resync: done=8, parse=8, recv=8
2016-04-27 21:11:04.438 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 P: got connection: 192.168.99.1:52240 -> 192.168.99.1:6432
2016-04-27 21:11:04.438 2549 NOISE safe_accept(8) = Resource temporarily unavailable
2016-04-27 21:11:04.438 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:04.439 2549 NOISE C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 pkt='!' len=84
2016-04-27 21:11:04.439 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 got var: user=postgres
2016-04-27 21:11:04.439 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 got var: database=postgres
2016-04-27 21:11:04.439 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 using application_name: psql
2016-04-27 21:11:04.439 2549 DEBUG C-0x7cbf60: (nodb)/(nouser)@192.168.99.1:52240 got var: client_encoding=UTF8
2016-04-27 21:11:04.439 2549 LOG C-0x7cbf60: postgres/postgres@192.168.99.1:52240 login attempt: db=postgres user=postgres tls=no
2016-04-27 21:11:04.439 2549 NOISE resync: done=84, parse=84, recv=84
2016-04-27 21:11:04.439 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:04.439 2549 NOISE C-0x7cbf60: postgres/postgres@192.168.99.1:52240 pkt='p' len=41
2016-04-27 21:11:04.439 2549 DEBUG finish_client_login: no welcome message, pause
2016-04-27 21:11:04.439 2549 DEBUG C-0x7cbf60: postgres/postgres@192.168.99.1:52240 pause_client
2016-04-27 21:11:04.439 2549 NOISE S-0x7d0c10: postgres/postgres@(bad-af):0 inet socket: 192.168.99.100
2016-04-27 21:11:04.439 2549 DEBUG S-0x7d0c10: postgres/postgres@192.168.99.100:9999 launching new connection to server
2016-04-27 21:11:04.439 2549 DEBUG launch_new_connection: already progress
2016-04-27 21:11:04.439 2549 DEBUG S-0x7d0c10: postgres/postgres@192.168.99.100:9999 S: connect ok
2016-04-27 21:11:04.439 2549 LOG S-0x7d0c10: postgres/postgres@192.168.99.100:9999 new connection to server (from 192.168.99.100:41466)
2016-04-27 21:11:04.439 2549 NOISE S-0x7d0c10: postgres/postgres@192.168.99.100:9999 P: SSL request
2016-04-27 21:11:04.439 2549 DEBUG launch_new_connection: already progress
2016-04-27 21:11:04.440 2549 NOISE resync: done=0, parse=0, recv=0
2016-04-27 21:11:04.440 2549 NOISE S-0x7d0c10: postgres/postgres@192.168.99.100:9999 launching tls
2016-04-27 21:11:04.441 2549 NOISE resync: done=1, parse=1, recv=1
2016-04-27 21:11:04.441 2549 NOISE tls_handshake: err=-2
2016-04-27 21:11:04.441 2549 DEBUG launch_new_connection: already progress
2016-04-27 21:11:04.442 2549 NOISE tls_handshake: err=-2
2016-04-27 21:11:04.442 2549 DEBUG launch_new_connection: already progress
2016-04-27 21:11:04.454 2549 NOISE tls_handshake: err=0
Segmentation fault
```

**Comment by theyetter** [comment]
I am witnessing this same behavior with:
pgpool-II version 3.5.4 (ekieboshi)
pgbouncer version 1.7.2

psql -> pgbouncer -> postgres works fine.
psql -> pgpool-ii -> postgres works fine.
psql -> pgbouncer -> pgpool-II -> postgres produces a segmentation fault as above.

**Comment by markokr** [comment]
Fixed in master, libusual commit e28d6a5
